sdk/src/trace/tracer.cc

```diff
@@ -51,6 +58,10 @@ nostd::shared_ptr<opentelemetry::trace::Span> Tracer::StartSpan(
   opentelemetry::trace::SpanContext parent_context = GetCurrentSpan()->GetContext();
   if (nostd::holds_alternative<opentelemetry::trace::SpanContext>(options.parent))
   {
+    if (!tracer_config_->IsEnabled())
+    {
+      return kNoopTracer->StartSpan(name, attributes, links, options);
+    }
```

Comment:
In my understanding, this should be moved earlier, not done only for the SpanContext alternative.

If the tracer is disabled, it is disabled for all types of spans.

Reply:
Ah yes, it should be. Missed this.